CIScripts/Common/Retry-Command.ps1

```diff
@@ -0,0 +1,32 @@
+function Retry-Command {
+    Param (
+        [Parameter(Mandatory = $true)] [int] $IntervalInSecs,
+        [Parameter(Mandatory = $true)] [int] $TimeoutInSecs,
+        [Parameter(Mandatory = $true,
+                   ValueFromPipeline=$true)] [ScriptBlock] $ScriptBlock
+    )
+    # Utility wrapper.
+    # Retries the specified command until it doesn't throw or
+    # timeouts. In case of timeout, an exception is raised. If it doesn't throw,
+    # command's output is returned.
+    #
+    # Example usage:
+    # { Get-AgentServiceStatus -Session $Session } `
+    #     | Retry-Command -IntervalInSecs 1 -TimeoutInSecs 5 `
+    #     | Should Be "Stopped"
+    #
+
+    $StartTime = Get-Date
+
+    while(((Get-Date) - $StartTime).Seconds -lt $TimeoutInSecs) {
+        try {
+            $ReturnVal = & $ScriptBlock
```

Comment:
Missing `Write-Host ""` before return (so the next printed line is not prepended by dots)